### ISSUE TITLE
Do we need to provide all variables with cssVars.variables? (IE11)

### ISSUE THREAD

**Issue by Selupsis** [jhildenbiddle/css-vars-ponyfill#99]
I have created a small **Angular** sample application including css-vars-ponyfill that I'm currently testing on IE11. It contains a title, a button to toggle the fontweight of the title, and a colorpicker to change the color of the title.

cssVars is initialized by calling:
```
cssVars({
   watch: true,
   onlyLegacy: true
})
```

The variables are:
```
--fontWeight: bold;
--fontColor: red;
```

In the methods to update the color or title I simply call e.g. ```cssVars(variables:{"--fontColor", "blue"});``` to update the color. Which works fine, **but** everytime I update one of the variables, the other gets reset to it's original value!

Is is necessary to call cssVars with variables containing **all** variables? This would be a bit of an overkill for larger applications...

**Comment by jhildenbiddle** [comment]
@Selupsis --

Confirmed 2.x bug. The behavior you are seeing is not intended.

I'll take a look before the end of the week. Thanks for filing the issue!

**Comment by Selupsis** [comment]
Thank you for the feedback, I'll wait for the fix then :)

**Comment by nkamenar** [comment]
Any update on this bug? Having to set all variables at once is making development very difficult.

**Comment by jhildenbiddle** [comment]
Apologies for the slow response, @Selupsis and @nkamenar. I recently relocated which has been keeping me busy.

I'll have some time to focus on this starting on Monday. This bug definitely needs to be fixed, so it is a priority.

**Comment by dimkadenisov** [comment]
By me script is not warking in ie 11. I have this error.
<img width="1920" alt="Screenshot 2019-12-03 at 23 45 43" src="https://user-images.githubusercontent.com/32514564/70088378-2fd14180-1627-11ea-89ed-2e17760ddb61.png">

**Comment by jhildenbiddle** [comment]
@dimkadenisov --

Please create a separate issue as yours is unrelated to this one.

**Comment by itsthechad** [comment]
I have a couple thoughts on this issue:

--------------

First, I believe I have a one-line fix for the bug. In [index.js](https://github.com/jhildenbiddle/css-vars-ponyfill/blob/master/src/index.js), it looks like the merging of variable stores simply isn't taking existing variables into account. The following change to line 396 seems to fix that:

Current: `Object.assign(jobVars, settings.variables);`
Proposed: `Object.assign(jobVars, variableStore.user, settings.variables);`

I've tested the above in my own project, and it resolves the issue. @jhildenbiddle, If this seems reasonable and you'd like me to create a PR, let me know.

--------------

Secondly, I have created a wrapper that acts as a temporary workaround for the issue. If my suggested fix above isn't acceptable or someone needs a fix immediately, the following may be helpful.

Wrapper gist: https://gist.github.com/itsthechad/c544df15c1031634363fc071084d911d
Codepen demo: https://codepen.io/itsthechad/pen/rNapZNQ

To use it, you can import as cssVars from the wrapper module rather than from css-vars-ponyfill. Then call `cssVars({ ...options })` just like you would call the ponyfill (same signature).

**Comment by Selupsis** [comment]
@itsthechad Thanks for the workaround, I can confirm that it works. I'll be using it until there will be a patch for this issue ^^

**Comment by itsthechad** [comment]
@jhildenbiddle, have you had a chance to look at my proposed solution above?

**Comment by itsthechad** [comment]
I went ahead and created a PR for this issue in hopes of getting it resolved: https://github.com/jhildenbiddle/css-vars-ponyfill/pull/110

(Please note this is my first github PR, so please let me know if I missed a step.)

**Comment by jhildenbiddle** [comment]
@Selupsis, @nkamenar, @itsthechad --

I have fixed this issue and will publish an update very soon.

For those interested, the issue was more complex than the workaround (#110) provided by @itsthechad would suggest. The workaround forced `options.variables` values to *always* persist, which breaks ponyfill calls made when `updateDOM` is `false`. This probably seems like an edge case, but it's critical for supporting web components since `updateDOM:false` is used under the hood to process shadowRoot `:root` declarations without affecting the regular DOM.

Thank you for your patience with this issue and apologies (again) for the slow response. Special thanks to @itsthechad providing the workaround while I was navigating the relocation!

**Comment by jhildenbiddle** [comment]
Fix is now available in 2.1.3
